apps/signal/56-forms-and-signal/src/app/order.component.ts

```diff
@@ -63,10 +64,24 @@ export default class OrderComponent {
   price = computed(
     () => products.find((p) => p.id === this.productId())?.price ?? 0,
   );
-  quantity = toSignal(this.form.controls.quantity.valueChanges, {
+  updatedQuantity = toSignal(this.form.controls.quantity.valueChanges, {
     initialValue: this.form.getRawValue().quantity,
   });
-  totalWihoutVat = computed(() => Number(this.price()) * this.quantity());
+  totalWihoutVat = computed(
+    () => Number(this.price()) * this.updatedQuantity(),
+  );
   vat = computed(() => this.totalWihoutVat() * 0.21);
   total = computed(() => this.totalWihoutVat() + this.vat());
+
+  quantity = input<number>();
+
+  constructor() {
+    effect(() => {
+      const q = this.quantity();
+
+      if (q) {
+        this.form.controls.quantity.patchValue(q);
+      }
+    });
+  }
```

Comment:
I did a similar solution using input signal. But I think I prefer the one of Billy https://github.com/tomalaforge/angular-challenges/pull/1044 using activatedRoute because the route is inialized before the component and you can initialize the form imediatly. Easier to understand and read.